server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -655,20 +655,27 @@ public void reset(CopyTo copyTo) {
          * @param defaultValue  the default value for the parameter, used if unspecified in mappings
          * @param parser        a function that converts an object to a parameter value
          * @param initializer   a function that reads a parameter value from an existing mapper
+         * @param serializer    a function that serializes a parameter value, prefer type specific x-content generation methods here for
+         *                      good performance as this is used on the hot-path during cluster state updates
+         * @param conflictSerializer a function that serializes a parameter value on conflict
```

Comment:
This is good, but I think a non-javadoc comment along the lines of 'DO NOT REPLACE THIS WITH A DEFAULT PARAMETER, IT WILL KILL PERFORMANCE' would also be helpful for future developers :)

Reply:
:D ok I tried my best to convey that message now

Reply:
Looks good, thanks

Reply:
Heya, my general question on this type of changes, besides leaving comments, are we going to catch it if we change this back in the future? Ideally we would have some tests fail but I can imagine that this is difficult? Are we going to see a regression in our benchmarks at least?

Reply:
I don't really see a way to create a test for this. Technically, we could improve upon the situation by removing the untyped methods from the x-content writer but that seems like a pretty big loss in comfort for more trivial usage of these APIs.

We are however working on periodic benchmarks for many-shards. This one I think would be potentially detectable (came out to a ~2-3% improvement in the time it takes us to bootstrap a 50k indices cluster).